src/ArbiterClientShim.cc

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2015-2017 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+#include <sched.h>
+#include "ArbiterClientShim.h"
+
+namespace Arachne {
+
+/**
+  * Block until setRequestedCores with a higher number causes a notification on
+  * this semaphor.
+  */
+core_t ArbiterClientShim::blockUntilCoreAvailable() {
+    inactiveCores.wait();
+    return sched_getcpu();
+}
+
+/**
+  * Returns true iff currentRequestedCores <= currentCores.
+  */
+bool ArbiterClientShim::mustReleaseCore() {
+    // Double-checked locking
```

Comment:
This comment is too terse. It may mean something to you, but I'm not sure it will mean anything to anyone else (is "double-checked locking" a term with well-defined meeting that you would expect all readers to know?). Let's discuss.